src/components/editor/editor-window/editor-window.tsx

```diff
@@ -14,22 +15,104 @@ import 'codemirror/addon/search/match-highlighter'
 import 'codemirror/addon/selection/active-line'
 import 'codemirror/keymap/sublime.js'
 import 'codemirror/mode/gfm/gfm.js'
-import React, { useState } from 'react'
+import 'codemirror/addon/hint/show-hint'
+import 'codemirror/addon/hint/show-hint.css'
+import { Data, NimbleEmojiIndex } from 'emoji-mart'
+import React, { useCallback, useState } from 'react'
 import { Controlled as ControlledCodeMirror } from 'react-codemirror2'
 import { useTranslation } from 'react-i18next'
 import './editor-window.scss'
 import { defaultKeyMap } from './key-map'
 import { ToolBar } from './tool-bar/tool-bar'
+import { addEmoji } from './tool-bar/utils'
+import data from 'emoji-mart/data/twitter.json'
 
 export interface EditorWindowProps {
   onContentChange: (content: string) => void
   content: string
 }
 
+interface findWOrdAtCursorResponse {
+  start: number,
+  end: number,
+  text: string
+}
+
+const allowedCharsInEmojiCodeRegex = /(:|\w)/
+
+const findWordAtCursor = (cm: Editor):findWOrdAtCursorResponse => {
```

Comment:
```suggestion
const findWordAtCursor = (editor: Editor): findWOrdAtCursorResponse => {
```

The Rest would need to change accordingly…